README-gen.md

```diff
@@ -0,0 +1,75 @@
+
+# Generalization
+
+There is an experimental new program `osm2pgsql-gen`. It is only built if the
+CImg and potrace libraries are available.
+
+This is part of a project to add generalization support to osm2pgsql. See
+https://osm2pgsql.org/generalization/ for more information.
+
+## Usage
+
+Run osm2pgsql as usual. This example assumes you have a table called
+`landcover` with a polygon geometry column called `geom` and a text column
+called `type` with the type of the landcover (forest, grass, etc.).
+
+Create a table `landcover_z10` with (at least) columns `geom` and `type` as
+well as integer columns `x`, and `y`. Then call this:
+
+```
+PGDATABASE=mydb ./osm2pgsql-gen -t landcover -T landcover_z10 -g geom -G type -z 10 -m 0.125 -s raster
+```
+
+The `landcover_z10` table will be filled with generalized polygons.
+
+Database connection parameters have to be specified using [environment
+variables](https://www.postgresql.org/docs/current/libpq-envars.html).
+
+Call `osm2pgsql-gen -h` for more command line options.
+
+## Extent
+
+You can process a single tile by setting `-x`, and `-y`. If `-X` and `-Y` are
+also set, all tiles between the arguments of `-x` and `-X`, and `-y` and `-Y`
+are processed. Without any of these, the program gets the extent from the data
+in the source table.
+
+You can also use the option `-e, --expire-list=TABLE` to read the list of tiles
+from a database table.
+
+In any case `-z` or `--zoom` sets the zoom level.
+
+## Parameters
+
+For a full list of parameters see `osm2pgsql-gen -h`.
+
+* `-s, --strategy=STRATEGY`: Set the strategy used for generalization. Currently
+  supported are `raster-union` and `vector-union`.
+* `-m, --margin=MARGIN`: This sets the margin around the tile as a fraction of
+  the tile size. So a value of 0.1 sets a 10% boundary on each side of the tile,
+  so as a result the tiles will overlap by 20% of their size. When using the
+  `raster-union` strategy the margin will be rounded up to the nearest multiple
+  of 64 pixels.
+* `-w, --width=WIDTH`: Size of the imaged rendered when using the `raster-union`
+  strategy, not used in the `vector-union` strategy.
+* `-b, --buffer=BUFFER`: The amount by which the polygons will be buffered. For
+  the `raster-union` strategy this is in pixels, for the `vector-union` strategy
+  this is in Mercator map units.
+* `-g, --group-by-column=COL`: Set this to the column describing the type of
```

Comment:
How are multiple columns indicated?

Reply:
That's not possible. If that's really something we need, we can add that later. But I am trying to keep the config to a minimum.